### PR TITLE
Mm-20223 Add webp emoji support

### PR DESCRIPTION
**Pull request by lapaz17**
#### Summary
This pr enables end users to upload webp files as an emoji. 
Note that since webp [library](https://pkg.go.dev/golang.org/x/image/webp) does not have encoding support, during tests I read existing webp file as suggested by @hmhealey in [this](https://community.mattermost.com/core/pl/p9xxjjqib38r8q955fz6fmgefa) reply.
#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/20223

#### Release Note

```release-note
Adds support for webp image types for emojis and image file uploads.
```

### CODE REVIEW

File: utils/emoji.go
Comment (agnivade) [inline utils/emoji.go:65]: There's no need to create a util function yet. We can write this in the caller function.

**Review by agnivade — COMMENTED** [review]
Thanks @lapaz17. However, this needs some additional changes elsewhere. Some client side changes are needed. Mainly the image picker dialogs for uploading emojis/images needs to be modified to say that we support webp images as well.

Also, the Mattermost API reference needs to be updated: https://github.com/mattermost/mattermost-api-reference/. Site is at: https://github.com/mattermost/mattermost-api-reference/.

File: api4/emoji_test.go
Comment (lapaz17) [reply api4/emoji_test.go:96]: It's already trying to upload webp image at line 101(createemoji calls DoEmojiUploadFile)? Could you elaborate your request if you asked for another thing?
image

File: utils/emoji.go
Comment (lapaz17) [reply utils/emoji.go]: Done.

**Comment by lapaz17** [comment]
> Thanks @lapaz17. However, this needs some additional changes elsewhere. Some client side changes are needed. Mainly the image picker dialogs for uploading emojis/images needs to be modified to say that we support webp images as well.
> 
> Also, the Mattermost API reference needs to be updated: https://github.com/mattermost/mattermost-api-reference/. Site is at: https://github.com/mattermost/mattermost-api-reference/.

@agnivade For the webapp, I modified translations (see [pr](https://github.com/mattermost/mattermost-webapp/pull/10439/files)) . But  for documentation I cannot find what to add? Seems like [emoji.yaml](https://github.com/mattermost/mattermost-api-reference/blob/0c90bd5890ccce2ef175d51c5b35e4020b0ff4cd/v4/source/emoji.yaml) has no mention about file types.

File: api4/emoji_test.go
Comment (agnivade) [reply api4/emoji_test.go:97]: I was referring to images being uploaded as post attachments. Those will support webp as well and we need to add tests for that. We can add another test case at `func TestUploadFiles` in api4/file_test.go.

**Comment by agnivade** [comment]
> But for documentation I cannot find what to add? Seems like [emoji.yaml](https://github.com/mattermost/mattermost-api-reference/blob/0c90bd5890ccce2ef175d51c5b35e4020b0ff4cd/v4/source/emoji.yaml) has no mention about file types.

Ok, then let's mention them. We can mention the same set of file types mentioned in the webapp.

**Comment by lapaz17** [comment]
> > But for documentation I cannot find what to add? Seems like [emoji.yaml](https://github.com/mattermost/mattermost-api-reference/blob/0c90bd5890ccce2ef175d51c5b35e4020b0ff4cd/v4/source/emoji.yaml) has no mention about file types.
> 
> Ok, then let's mention them. We can mention the same set of file types mentioned in the webapp.

Opened [pr](https://github.com/mattermost/mattermost-api-reference/pull/704/files).

File: api4/emoji_test.go
Comment (lapaz17) [reply api4/emoji_test.go:97]: Just added.

**Review by agnivade — APPROVED** [review]
Thanks

**Comment by hmhealey** [comment]
@agnivade Do we still need the Do Not Merge label for this? This one seems like the PR that's blocking the others at this point

**Comment by agnivade** [comment]
This PR is the main PR that needs to go in along with the translation PR. So yes, I think we need the label for this. The webapp PR doesn't make sense to go in without server support.

**Comment by agnivade** [comment]
Oh yeah this can be merged first, but it needs to go in with the doc updates. The "do not merge" was meant for the doc update PR. No worries with the doc updates as they are good to go.

So as soon as PM and QA review is complete, we can merge this and https://github.com/mattermost/mattermost-api-reference/pull/704.

**Comment by agnivade** [comment]
Gentle ping @esethna and @jgilliam17 on this.

**Comment by jgilliam17** [comment]
Thanks @lapaz17 
Tested, looks good to merge - pending E2E report (in progress)
- Verified/able to upload webp files as an emoji.

**Comment by jgilliam17** [comment]
E2Es are failing to start, removing the `do not merge` label and adding back after successful cycle starts.

**Comment by agnivade** [comment]
@esethna - can you take a quick look and let us know if there are any concerns from your side?

**Review by esethna — APPROVED** [review]
Sorry for the delay, no issues from my end

**Pull request merged**
